tickets/views.py

```diff
@@ -400,6 +401,7 @@ def moderation_assigned(request, user_id):
                     if ticket.sound.pack:
                         packs_to_update.add(ticket.sound.pack)
                 Sound.objects.filter(ticket__in=tickets).delete()
+                tickets.update(sound_id=None)
```

Comment:
`Ticket` model has a foreign key to `Sound` [with `on_delete=models.SET_NULL`](https://github.com/MTG/freesound/blob/master/tickets/models.py#L58), therefore the manual update of the field should not really be necessary right? Did you understand why this is happening? Isn't the `transaction.atomic()` bit already fixing the problem?

Reply:
Note that the error was in line https://github.com/MTG/freesound/pull/947/files#diff-f0c47805ccfcbaead49a83d1ce646e68R437. So I think the problem was that the local objects either were not refreshed with the data in the database or they were modified after.
Now, by calling update on the queryset the values are updated immediately for all the Ticket.

Reply:
I just checked that only using `transaction.atomic()` doesn't fix the problem

Reply:
I see, makes sense :)

Reply:
Then we should be using maybe something like `obj.refresh_from_db()` but for a queryset right? (if it exists)

Reply:
In any case, the solution seems ok but a comment should be left there explaining why we need to explicitly call `tickets.update(sound_id=None)`

Reply:
After talking this through with Frederic I finally understand the issue - I didn't realise that it happened in the post_save trigger for `TicketComment`. My preference here would be to use `Ticket.objects.filter` again instead of the `update`. Perhaps with a comment like

    # After we delete sounds that these tickets are associated with, we refresh the ticket list so that sound_id is null and this does not affect the TicketComment post_save trigger

I think that the comment is important because we both looked at the code and intuitively thought that the update wasn't needed because of the `on_delete` setting. Having an explicit explanation is a good idea

Reply:
I changed it for refresh_from_db as @ffont suggested. If you think is still not clear I can change it